src/packet/process_result.rs

```diff
@@ -0,0 +1,102 @@
+use crate::either::Either;
+use crate::packet::{OutgoingPacket, Packet, PacketType};
+
+use std::collections::VecDeque;
+/// Helper class that implements `Iterator`, and is used to return incoming (from bytes to packets) or outgoing (from packet to bytes) packets.
+/// It is used as optimization in cases, where most of the time there is only one element to iterate, and we don't want to create a vector for it
+pub struct ZeroOrMore<T> {
```

Comment:
this is only used here internally right, we can remove the pub

Reply:
We can't, it is used in implementation of `IntoIterator` trait, as an associated type.